modules/light-clients/08-wasm/keeper/msg_server_test.go

```diff
@@ -278,3 +279,102 @@ func (suite *KeeperTestSuite) TestMsgMigrateContract() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestMsgRemoveCodeHash() {
+	codeHash := sha256.Sum256(wasmtesting.Code)
+
+	govAcc := authtypes.NewModuleAddress(govtypes.ModuleName).String()
+
+	var (
+		msg           *types.MsgRemoveCodeHash
+		expCodeHashes []types.CodeHash
+		expFound      bool
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {
+				msg = types.NewMsgRemoveCodeHash(govAcc, codeHash[:])
+
+				expCodeHashes = []types.CodeHash{}
+				expFound = true
+			},
+			nil,
+		},
+		{
+			"success: many code hashes",
+			func() {
+				msg = types.NewMsgRemoveCodeHash(govAcc, codeHash[:])
+
+				expFound = true
+				expCodeHashes = []types.CodeHash{}
+
+				for i := 0; i < 20; i++ {
+					codeHash := sha256.Sum256([]byte{byte(i)})
+					err := ibcwasm.CodeHashes.Set(suite.chainA.GetContext(), codeHash[:])
+					suite.Require().NoError(err)
+
+					expCodeHashes = append(expCodeHashes, codeHash[:])
+				}
+			},
+			nil,
+		},
+		{
+			"success: code hash is missing",
+			func() {
+				msg = types.NewMsgRemoveCodeHash(govAcc, []byte{1})
+
+				expCodeHashes = []types.CodeHash{codeHash[:]}
+				expFound = false
+			},
+			nil,
+		},
+		{
+			"failure: unauthorized signer",
+			func() {
+				msg = types.NewMsgRemoveCodeHash(suite.chainA.SenderAccount.GetAddress().String(), codeHash[:])
+			},
+			ibcerrors.ErrUnauthorized,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupWasmWithMockVM()
+
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
+
+			tc.malleate()
+
+			ctx := suite.chainA.GetContext()
+			res, err := GetSimApp(suite.chainA).WasmClientKeeper.RemoveCodeHash(ctx, msg)
+			events := ctx.EventManager().Events().ToABCIEvents()
+
+			if tc.expError == nil {
+				suite.Require().NoError(err)
+				suite.Require().NotNil(res)
+				suite.Require().Equal(expFound, res.Found)
+
+				codeHashes, err := types.GetAllCodeHashes(suite.chainA.GetContext())
+				suite.Require().NoError(err)
+
+				// Check equality of code hashes up to order
+				suite.Require().Subset(expCodeHashes, codeHashes)
+				suite.Require().Subset(codeHashes, expCodeHashes)
+
+				// Verify events
+				suite.Require().Len(events, 0)
```

Comment:
I wouldn't add this necessarily but if we do we can atleast move getting the events in the successful case.

Reply:
You think we should emit events?

Reply:
is there actually 0 events emitted? hmm, would've thought sdk baseapp would emit some std ones

Reply:
The context is not going through the baseapp. I'm directly calling the msg_server handler with the context

Reply:
makes sense!

Reply:
not sure re events. I dont think they'd hurt tho.